vault/testing.go

```diff
@@ -266,6 +279,60 @@ func testTokenStore(t testing.TB, c *Core) *TokenStore {
 	return ts
 }
 
+// TestIdentityStoreWithGithubAuth returns an instance of identity store which
+// is mounted by default. This function also enables the github auth backend to
+// assist with testing identities and entities what require an valid mount ID
+// of an auth backend.
+func TestIdentityStoreWithGithubAuth(t testing.TB) *identityStore {
```

Comment:
OK, get ready to hate me, but I don't really like having GitHub used for this as it requires a token to be set or the test doesn't run. I think we'd be better off using LDAP with the public test server that the existing LDAP test code uses.